service/cachemdw/cache_test.go

```diff
@@ -103,12 +105,27 @@ func TestUnitTestCacheQueryResponse(t *testing.T) {
 	resp, err = serviceCache.GetCachedQueryResponse(ctxb, req)
 	require.NoError(t, err)
 	require.JSONEq(t, string(defaultQueryResp), string(resp.JsonRpcResponseResult))
+
+	// same request with different ids should return same cached response, but with correct id
+	stringId := "this is a string id"
+	req = mkEVMRPCRequestEnvelope(defaultBlockNumber, stringId)
+	resp, err = serviceCache.GetCachedQueryResponse(ctxb, req)
+	require.NoError(t, err)
+	expectedRes := strings.Replace(string(defaultQueryResp), "\"id\": 1", fmt.Sprintf("\"id\": \"%s\"", stringId), 1)
+	require.JSONEq(t, expectedRes, string(resp.JsonRpcResponseResult))
+
+	boolId := true
```

Comment:
A bool isn't actually allowed by the json rpc 2.0 spec

Reply:
ha may not be in the spec but ethermint seems to support it. not suggesting it _has_ to. just something i noticed and i wanted to try out any arbitrary id that worked.

good call on `null`. without the proxy it properly returns an `id` of `null`. but the proxy converts it to `0`. to properly relay `null` i think i need to make `ID` a pointer (`*interface{}` instead of `interface{}`)

Reply:
jk 🤦 i was testing against live which has ID as an `int64`. updated this test to be for `nullId` instead of `boolId`

Reply:
ha thnx for adding tests.  Ya, etheremint probably should validate that (I believe geth does)